### ISSUE TITLE
M+ Stories Staging Website

### ISSUE THREAD

**Issue by micahwalter** [mplusmuseum/mplusmuseum-stories#185]

**Comment by micahwalter** [comment]
- [ ] Clone DB to new "staging" db
- [ ] Restore ELB for staging CMS
- [ ] Update config.php to point at new DB

**Comment by micahwalter** [comment]
@chrismplus @rdunk - OK this is done. There is now a "staging" database. It was a clone of production as of last week. Please note these two databases wont stay in sync any further, but we can re-clone production in the future when needed. 

To use the staging CMS, just go to here https://staging-cms.mplus.org.hk/admin/ -- 

I am fixing up a few things for the actual front end site on staging now as well...

**Comment by chrismplus** [comment]
@micahwalter — thanks. So the front end of Staging will eventually be available here: https://staging-cms.mplus.org.hk/ ?

**Comment by micahwalter** [comment]
@chrismplus - 

* Staging site for the CMS is here https://staging-cms.mplus.org.hk - This uses the clones DB, separate of the live CMS DB.
* Staging site for the front-end will be here https://staging-stories.mplus.org.hk - This will be pointed at staging-cms / If we ever want we can switch this to point at live data to test any major code updates. - I'm still working on getting this to work properly.

**Comment by chrismplus** [comment]
Thanks @micahwalter — can you let me know how things are sitting with this?

**Comment by micahwalter** [comment]
➤ Micah Walter commented:

I just added Eric to this task. Basically it's all completed, but there is some issue connecting the Stories staging front end to the Staging CMS backend. We will have a look shortly.

**Comment by chrismplus** [comment]
Thanks, @micahwalter, @erchaves — is there any update on this?

**Comment by chrismplus** [comment]
Hi @micahwalter — just looping back on this one. Is this something that we can push through shortly? We'd like to prototype some new sections for Stories on a Staging environment. It's difficult to do on Production. 

CC: @dianewang @laramplus

**Comment by micahwalter** [comment]
@chrismplus @rdunk -- I'll have to revisit this a bit more but it looks like the staging site is currently working here https://staging-stories.mplus.org.hk/

**Comment by chrismplus** [comment]
@micahwalter — the staging site is available but we cannot login. This is where the issue has been left, connecting the Stories staging front-end fo the Staging CMS backend.

**Comment by erchaves** [comment]
I left @chrismplus a slack message about this as well...
but when i log in to the CMS at https://cms.mplus.org.hk/ I just see this empty page. 
![screen shot 2018-09-19 at 4 04 15 pm](https://user-images.githubusercontent.com/302309/45781633-8c1c9100-bc2e-11e8-93ba-37e33502f92c.png)

And none of the buttons do anything. “Continue” and “view site” just reload the same page. Am I missing something?

The same happens on both staging and production. My same log in works for both, but i can’t see anything besides that page.

**Comment by chrismplus** [comment]
Hi @erchaves — I've Slacked you an update on this. All should be working now.

**Comment by chrismplus** [comment]
Hi @erchaves — let me know whether you need anything else for this?

**Comment by chrismplus** [comment]
Hi @erchaves — logging this here as it seems to be a final bug (but crucial to the point of a staging site). Whenever a new image is uploaded on the staging site it doesn't render on the front end. 

Please refer to image below. @rdunk — do you know why this might be happening?
![image](https://user-images.githubusercontent.com/26731347/49640063-883b2680-fa47-11e8-91cc-692cb647ba6c.png)

**Comment by erchaves** [comment]
Okay thanks @chrismplus good catch. @rdunk if you have any insight that would be great, or no worries if you're not sure. I can look into this when i get a chance next week. Apologies for the slow turn around on these as we're pretty heads down on the collection explorer. But I'll put it in my queue.

**Comment by chrismplus** [comment]
Thanks, @erchaves — @rdunk can see whether you have any insight into what might be happening on this one?

**Comment by rdunk** [comment]
I suspect this is CDN related. Not sure that the CDN caches content from the staging instance, in fact I doubt it does... I'll try disabling...

**Comment by erchaves** [comment]
Oh, that's great, thanks @rdunk! I've been tied up with the collection explorer stuff and hadn't had a chance to look at this. Does that mean we can close this task now? Thanks for looking into that.

**Comment by chrismplus** [comment]
Thanks. Verified it works now. Closing.

**Comment by chrismplus** [comment]
@rdunk — there are still issues with this. For whatever reason the image upload still returns an error: 
![image](https://user-images.githubusercontent.com/26731347/50273658-31d1dd00-0476-11e9-8e7f-46784ac9b725.png)

I'll Slack you a link to preview...

**Comment by chrismplus** [comment]
The View tab on the Staging Site is also reverting back to the Production instance again

**Comment by micahwalter** [comment]
OK, this is finally resolved. We have a staging CMS with its own database, and a staging front end running on Heroku.

**Comment by chrismplus** [comment]
Hi @rdunk, @micahwalter — changes made on the Staging Site is flowing through to our Production Site. I unpublished _[Which Old-School Emoji Are You?](https://stories.mplus.org.hk/en/blog)_ on Staging to test @rdunk's deployment of the Scheduling Content #156 — but this also unpublished the blog post on the Production. 

@rdunk — when I unhid the blog on Staging it flowed back through to the main homepage banner on Production but didn't reappear on the Blog Roll. I had to republish the entire Blog node to fix this. 

Staging seems very unstable. I'd like to ensure that whatever happens there doesn't affect Production. Do we have a roadmap for fixing these on-going issues? We seem to be resurfacing a lot of old bugs. Let me know whether you have any thoughts.
